src/main/java/org/wise/portal/presentation/web/controllers/peergroup/PeerGroupCreateController.java

```diff
@@ -0,0 +1,46 @@
+package org.wise.portal.presentation.web.controllers.peergroup;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.wise.portal.domain.group.impl.PersistentGroup;
+import org.wise.portal.domain.peergroup.PeerGroup;
+import org.wise.portal.domain.peergroupactivity.PeerGroupActivity;
+import org.wise.portal.domain.run.impl.RunImpl;
+import org.wise.portal.service.peergroup.PeerGroupCreateService;
+import org.wise.portal.service.peergroupactivity.PeerGroupActivityNotFoundException;
+import org.wise.portal.service.peergroupactivity.PeerGroupActivityService;
+import org.wise.portal.service.run.RunService;
+
+@RestController
+@Secured("ROLE_TEACHER")
+@RequestMapping("/api/peer-group/create")
+public class PeerGroupCreateController {
+
+  @Autowired
+  private PeerGroupActivityService peerGroupActivityService;
+
+  @Autowired
+  private PeerGroupCreateService peerGroupCreateService;
+
+  @Autowired
+  private RunService runService;
+
+  @PostMapping("/{runId}/{periodId}/{nodeId}/{componentId}")
+  PeerGroup create(@PathVariable("runId") RunImpl run,
+      @PathVariable("periodId") PersistentGroup period, @PathVariable String nodeId,
+      @PathVariable String componentId, Authentication auth)
+      throws PeerGroupActivityNotFoundException {
+    if (runService.hasWritePermission(auth, run)) {
+      PeerGroupActivity activity =
+          peerGroupActivityService.getByComponent(run, nodeId, componentId);
+      return peerGroupCreateService.create(activity, period);
+    }
+    throw new AccessDeniedException("Not permitted");
+  }
```

Comment:
I noticed that you can create a Peer Group for a run and period combination where the period is not actually in that run but in a different run.

We may want to add a check to make sure the period that is passed as a parameter is actually in the run that is passed in as a parameter.

Reply:
Good catch. I added a test to make sure that the requested period belongs in the requested run.